reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt

```diff
@@ -321,278 +310,308 @@ class StaticHtmlReporter : Reporter() {
+
+        table("ort-report-table ort-packages $excludedClass") {
+            thead {
+                tr {
+                    th { +"Package" }
+                    th { +"Scopes" }
+                    th { +"Declared Licenses" }
+                    th { +"Detected Licenses" }
+                    th { +"Analyzer Errors" }
+                    th { +"Scanner Errors" }
+                }
+            }
 
-                    append("""
-                        <tr class="$cssClass$rowExcludedClass">
-                            <td>${row.id}</td>
-                            <td>${formatScopes(row.scopes)}</td>
-                            <td><ul>
-                                ${row.declaredLicenses.joinToString("\n") {
-                                    "<li>$it</li>"
-                                }}
-                            </ul></td>
-                            <td><ul>
-                                ${row.detectedLicenses.joinToString("\n") {
-                                    "<li>$it</li>"
-                                }}
-                            </ul></td>
-                            <td><ul>
-                                ${row.analyzerErrors.joinToString("\n") {
-                                    if (it.isResolved) {
-                                        "<li class=\"ort-resolved\"><p>${it.description.replace("\n", "<br/>")}</p>" +
-                                                "<p>${it.resolutionDescription}</p></li>"
-                                    } else {
-                                        "<li><p>${it.description.replace("\n", "<br/>")}</p></li>"
-                                    }
-                                }}
-                            </ul></td>
-                            <td><ul>
-                                ${row.scanErrors.joinToString("\n") {
-                                    if (it.isResolved) {
-                                        "<li class=\"ort-resolved\"><p>${it.description.replace("\n", "<br/>")}</p>" +
-                                                "<p>${it.resolutionDescription}</p></li>"
-                                    } else {
-                                        "<li><p>${it.description.replace("\n", "<br/>")}</p></li>"
+            tbody { table.rows.forEach { projectRow(it) } }
+        }
+    }
+
+    private fun TBODY.projectRow(row: ReportTableModel.DependencyRow) {
+        // Only mark the row as excluded if all scopes the dependency appears in are excluded.
+        val rowExcludedClass =
+            if (row.scopes.isNotEmpty() && row.scopes.all { it.value.isNotEmpty() }) "ort-excluded" else ""
+
+        val cssClass = when {
+            row.analyzerErrors.containsUnresolved() || row.scanErrors.containsUnresolved() -> "ort-error"
+            row.declaredLicenses.isEmpty() && row.detectedLicenses.isEmpty() -> "ort-warning"
+            else -> "ort-success"
+        }
+
+        tr("$cssClass $rowExcludedClass") {
```

Comment:
The space between these variables seems to cause the trailing space in the CSS class that I mentioned earlier. The original code does not have this space.

Reply:
Yes, before the `rowExcludedClass` was set to `" ort-excluded"` with a leading space and the final string was built using `"$cssClass$rowExcludedClass"`. I didn't like that either and decided to have the trailing space instead, because the same pattern is used elsewhere and could lead to a leading space instead (which doesn't happen in the test file). I think that's a tiny detail that can be fixed later.

Reply:
Agreed. TBH, I wasn't fully aware anymore that you could have multiple (space-separated) class names listed in the `class` attribute, thus a (trailing or leading) space does not harm, as the quoted string is not interpreted as a literal string containing spaces.